src/util/debug.h

```diff
@@ -0,0 +1,47 @@
+
+#include <stdio.h>
+
+#include <HardwareSerial.h>
+
+#ifndef DEBUG_MODULE
+#error "Cannot include debug.h without defining DEBUG_MODULE"
+#endif
+
+#ifndef DEBUG_BUFFER_SIZE
+#define DEBUG_BUFFER_SIZE 128
+#endif
+
+#ifndef DEBUG_SERIAL_PORT
+#define DEBUG_SERIAL_PORT Serial
+#endif
+
+// Initial setup of debug; only to be used once
+#define DEBUG_BEGIN DEBUG_SERIAL_PORT.begin(115200)
+
+#ifdef DEBUG
+
+// Use DEBUG_PRINT to add print messages like printf
+#define DEBUG_PRINT(...)                              \
+  do {                                                \
+    char _buf[DEBUG_BUFFER_SIZE];                     \
+    snprintf(_buf, DEBUG_BUFFER_SIZE, __VA_ARGS__);   \
+    DEBUG_SERIAL_PORT.print("[" DEBUG_MODULE "] ");        \
```

Comment:
Will the `Serial.print()` command recognize the combining of the `"["` with `DEBUG_MODULE` and `"]"`?

Reply:
It doesnt have to; adjacent string literals and merged into one string at compile time in C (https://stackoverflow.com/questions/12120944/how-does-concatenation-of-two-string-literals-work), so after preprocessing, this looks like `"[" "main" "] "` which becomes `"[main] "`

Reply:
Ok cool just wondering! Everything else looks good!